### PR TITLE
Move validator-extends to it's own file

### LINKED ISSUE #4085: Proposal: move Validator with it's extends to own file
Just a short question, before starting a pull request: what do you think about moving all the Validator.extend() related stuff in [instance-validator.js](https://github.com/sequelize/sequelize/blob/master/lib/instance-validator.js) to it's own file (validator-extends.js), so it can be imported from the outside world?

### PR DESCRIPTION
**Pull request by gustavpursche**
This pull requests moves the validator.js code in `instance-validator.js` to `utils/validator-extends.js` to have the ability to reuse the manual extensions, sequelize makes, outside of the lib.

Thank's a lot for the great contribution docs!

Closes #4085

### CODE REVIEW

File: lib/utils/validator-extends.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var validator = require('validator')
+  , _ = require('lodash');
+
+var extensions = {
+      notEmpty: function(str) {
```
Comment (mickhansen) [inline lib/utils/validator-extends.js]: spacing issues

**Comment by mickhansen** [comment]
Spacing issues in the entirety of the one file.
Please name the file 'validator-extras'

File: lib/utils/validator-extends.js (hunk 1 above)
Comment (gustavpursche) [inline lib/utils/validator-extends.js]: What's the issue? No Space between `:` and `function`?

File: lib/utils/validator-extras.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        return str.indexOf(elem) >= 0 && !!elem;
+      },
+      notContains: function(str, elem) {
+        return !this.contains(str, elem);
+      },
+      is: function(str, pattern, modifiers) {
+        return this.regex(str, pattern, modifiers);
+      },
+    }
+  , extendModelValidations = function(modelInstance) {
```
Comment (mickhansen) [inline lib/utils/validator-extras.js]: define this seperately so the methods before can have a ltitle saner indentation for all that code :)

**Comment by gustavpursche** [comment]
Is there something missing here? Travis failure seems to be unrelated to this pull request ...

**Comment by janmeier** [comment]
Rebased and merged in https://github.com/sequelize/sequelize/commit/b0037d67f05a760476b76d2f0a1d2db75e5b58bf

**Comment by janmeier** [comment]
Thanks for your work @gustavpursche  :)

**Comment by janmeier** [comment]
Yeah, the travis failure is related to problems with the mssql servers

**Pull request closed without merging**